commons/all/src/main/java/org/infinispan/commons/api/CacheContainerAdmin.java

```diff
@@ -119,4 +119,19 @@ public static AdminFlag valueOf(int index) {
     * @return
     */
    C withFlags(EnumSet<AdminFlag> flags);
+
+   /**
+    *  Creates a template on the container using the provided configuration.
+    *
+    * @param name the name of the template
+    * @param configuration the configuration to use. It must be a clustered configuration (e.g. distributed)
```

Comment:
What's the reason to not allow a LOCAL configuration?

I was thinking of deprecating `DefaultCacheManager.defineConfiguration()` and telling the users to use the admin interface instead, but we can't do that if we don't allow LOCAL templates.

Reply:
I was simply replicating what we allow for `createCache`. I guess the idea is that a local cache/template should always be local and not replicated to other nodes. For example, if two nodes wanted a local cache with the same name.